test/phpunit/lib/filter/QubitLimitIpTest.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @internal
+ *
+ * @coversNothing
+ */
+class QubitLimitIpTest extends TestCase
+{
+    protected $contextMock;
+    protected $filter;
+
+    protected function setUp(): void
+    {
+        $configurationClass = 'qubitConfiguration';
+        if (!class_exists($configurationClass)) {
+            $this->markTestSkipped("The application configuration class '{$configurationClass}' does not exist.");
+        }
+
+        // Create an instance of the application configuration
+        $configuration = new $configurationClass('test', false);
+
+        // Create a sfContext instance
+        $this->contextMock = sfContext::createInstance($configuration);
+
+        // Create the filter instance and pass the sfContext
+        $this->filter = new QubitLimitIpFilter($this->contextMock);
+    }
+
+    public function testBypassLimitInDebugMode()
+    {
+        $debug = true;
+        $limit = explode(';', '192.168.0.1');
+        $firstCall = true;
+        $logout = ['user', false];
+
+        // Mock sfConfig to enable limit ip setting
+        sfConfig::set('app_limit_admin_ip', $limit);
+
+        // Call the method
+        $actual = $this->filter->shouldByPassLimit($debug, $limit, $firstCall, $logout);
+
+        // Assert that the limit is bypassed
+        $this->assertTrue($actual);
+    }
+
+    public function testBypassLimitEmptyLimit()
+    {
+        $debug = false;
+        $limit = explode(';', '');
+        $firstCall = true;
+        $logout = ['user', false];
+
+        // Mock sfConfig to enable limit ip setting
+        sfConfig::set('app_limit_admin_ip', $limit);
+
+        // Call the method
+        $actual = $this->filter->shouldByPassLimit($debug, $limit, $firstCall, $logout);
+
+        // Assert that the limit is bypassed
+        $this->assertTrue($actual);
+    }
+
+    public function testBypassLimitNotFirstCall()
+    {
+        $debug = false;
+        $limit = explode(';', '192.168.0.1');
+        $firstCall = false;
+        $logout = ['user', false];
+
+        // Mock sfConfig to enable limit ip setting
+        sfConfig::set('app_limit_admin_ip', $limit);
+
+        // Call the method
+        $actual = $this->filter->shouldByPassLimit($debug, $limit, $firstCall, $logout);
+
+        // Assert that the limit is bypassed
+        $this->assertTrue($actual);
+    }
+
+    public function testBypassLimitUserLogout()
+    {
+        $debug = false;
+        $limit = explode(';', '192.168.0.1');
+        $firstCall = true;
+        $logout = ['user', true];
```

Comment:
How does this test work? The action being sent is the bool value 'true' - this will not match 'user', 'OIDC' etc but it appears shouldByPassLimit is still returning `true`.